src/test/java/org/apache/sling/commons/mime/internal/MimeTypeServiceImplTest.java

```diff
@@ -152,6 +165,29 @@ public void testProvider2() throws Exception {
         assertNull(this.service.getMimeType(GIF));
     }
 
+    public void testMultipleMimeTypes() throws Exception {
+        loadMimeTypes(MimeTypeServiceImpl.CORE_MIME_TYPES);
+        loadMimeTypes(MimeTypeServiceImpl.MIME_TYPES);
+
+        for (String mm : epsMimeTypeExt.keySet()) {
+            assertEquals("Extension " + mm + " (1)", epsMimeTypeExt.get(mm), this.service.getExtension(mm));
+        }
+    }
+
+    public void testNegativeTests() throws Exception {
+        loadMimeTypes(MimeTypeServiceImpl.CORE_MIME_TYPES);
+        loadMimeTypes(MimeTypeServiceImpl.MIME_TYPES);
+
+        // null
+        String mm = null;
+        assertEquals("Extension " + mm, mm, this.service.getExtension(mm));
+        this.service.registerMimeType("application/invalid-1", mm);
+        mm = "";
+        assertEquals("Extension " + mm, null, this.service.getExtension(mm));
+        this.service.registerMimeType("application/invalid-1", mm);
```

Comment:
Why are these two registration methods needed? It is unclear what the test is validating, since you have assertions and then apparently an expected exception.

Reply:
They are negative tests with invalid values registered. Simplified the tests.